x/mongo/driver/compression_test.go

```diff
@@ -40,3 +43,78 @@ func TestCompression(t *testing.T) {
 		})
 	}
 }
+
+func TestZstdWindowSize(t *testing.T) {
+	tests := []struct {
+		inputSize  int
+		level      zstd.EncoderLevel
+		windowSize int
+	}{
+		{
+			inputSize:  512,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 1024,
+		},
+		{
+			inputSize:  512000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 524288,
+		},
+		{
+			inputSize:  16000000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 16777216,
+		},
+		{
+			inputSize:  32000000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 16777216,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(strconv.Itoa(test.inputSize), func(t *testing.T) {
+			windowSize := calcZstdWindowSize(test.inputSize, test.level)
+			assert.Equal(t, test.windowSize, windowSize)
+		})
+	}
+}
+
+func BenchmarkCompression(b *testing.B) {
+	payload := func() []byte {
+		f, err := os.Open("compression.go")
+		if err != nil {
+			b.Error(err)
+		}
+		buf, err := io.ReadAll(f)
+		if err != nil {
+			b.Error(err)
+		}
+		for i := 1; i < 10; i++ {
+			buf = append(buf, buf...)
+		}
+		return buf
+	}()
+
+	compressors := []wiremessage.CompressorID{
+		wiremessage.CompressorSnappy,
+		wiremessage.CompressorZLib,
+		wiremessage.CompressorZstd,
+	}
+
+	for _, compressor := range compressors {
+		b.Run(strconv.Itoa(int(compressor)), func(b *testing.B) {
```

Comment:
[nit] Looks like the benchmark name will just be the number of the compressor in `iota` order. Could we use the actual names of the compressors instead (might have to make this table-driven).